src/libraries/System.Diagnostics.TraceSource/src/System.Diagnostics.TraceSource.csproj

```diff
@@ -29,11 +29,11 @@
     <Compile Include="System\Diagnostics\SwitchLevelAttribute.cs" />
   </ItemGroup>
   <ItemGroup>
+    <Reference Include="System.IO.FileSystem" />
```

Comment:
This works? When I tried I got errors about not being able to find File. But if it works, great.

Reply:
i just tried the individual build and it works, let me check with the entore vertical build
```c#
C:\git\runtime\src\libraries\System.Diagnostics.TraceSource\src>dotnet build
Microsoft (R) Build Engine version 16.7.0-preview-20310-07+ee1c9fd0c for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  System.Private.CoreLib -> C:\git\runtime\artifacts\bin\coreclr\Windows_NT.x64.Debug\IL\System.Private.CoreLib.dll
  System.Collections -> C:\git\runtime\artifacts\bin\System.Collections\net5.0-Debug\System.Collections.dll
  System.Collections.NonGeneric -> C:\git\runtime\artifacts\bin\System.Collections.NonGeneric\net5.0-Debug\System.Collections.NonGeneric.dll
  System.Collections.Specialized -> C:\git\runtime\artifacts\bin\System.Collections.Specialized\net5.0-Debug\System.Collections.Specialized.dll
  System.ComponentModel.Primitives -> C:\git\runtime\artifacts\bin\System.ComponentModel.Primitives\net5.0-Debug\System.ComponentModel.Primitives.dll
  System.Private.Uri -> C:\git\runtime\artifacts\bin\System.Private.Uri\net5.0-Debug\System.Private.Uri.dll
  System.Runtime -> C:\git\runtime\artifacts\bin\System.Runtime\net5.0-Debug\System.Runtime.dll
  System.Runtime.Extensions -> C:\git\runtime\artifacts\bin\System.Runtime.Extensions\net5.0-Debug\System.Runtime.Extensions.dll
  System.Threading -> C:\git\runtime\artifacts\bin\System.Threading\net5.0-Debug\System.Threading.dll
  System.Diagnostics.TraceSource -> C:\git\runtime\artifacts\bin\System.Diagnostics.TraceSource\net5.0-Debug\System.Diagnostics.TraceSource.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:12.92
```

Reply:
The only type this assembly uses from FileSystem is `File` which is not defined in CoreLib.  If you were seeing errors for missing File it was because you didn't add the simple reference but removed the ProjectReference.
This would only be problematic if this assembly needed types which were defined in CoreLib, like FileStream.  In that case we would see duplicate types.

Reply:
I definitely added the Reference. But maybe something else went awry